### PR TITLE
Terraform 0.12

### PR DESCRIPTION
**Pull request by shaunc**
Update to terraform 0.12 -- primarily based on [sagottaros' fork](https://github.com/sagittaros/etcd-cloud-operator) -- but also merging latest additions to your master.

### CODE REVIEW

File: terraform/modules/ignition/ignition.tf [hunk 1]
```diff
@@ -14,45 +14,47 @@
...
+  users = [data.ignition_user.core.id]
 
   append {
-    source       = "${lookup(var.ignition_extra_config, "source", local.blank_ignition_config)}"
-    verification = "${lookup(var.ignition_extra_config, "verification", "")}"
+    source = lookup(
+      var.ignition_extra_config,
+      "source",
+      local.blank_ignition_config,
+    )
```
Comment (Quentin-M) [inline terraform/modules/ignition/ignition.tf]: nit: this lookup got split in several lines, while the other stayed a one liner

File: terraform/modules/ignition/ignition.tf [hunk 2]
```diff
@@ -66,86 +68,88 @@ data "ignition_systemd_unit" "update-engine" {
...
   name    = "node-exporter.service"
-  content = "${file("${path.module}/resources/node-exporter.service")}"
+  content = file("${path.module}/resources/node-exporter.service")
 }
 
 data "ignition_file" "eco-config" {
   filesystem = "root"
   path       = "/etc/eco/config.yaml"
-  mode       = 0644
+  mode       = 420
```
Comment (Quentin-M) [inline terraform/modules/ignition/ignition.tf]: nit: did they deprecate this syntax? it feels easier to read as 0644 to me as it is the usual chmod syntax.

**Comment by Quentin-M** [comment]
Pretty intense PR, thanks a lot Shaun!
We are going through the same process internally, but haven't gotten to ECO (not ignition!) yet.
Have you been able to test this? In which case, I'd merge it right away.

**Comment by shaunc** [comment]
So have ... lightly tested it (does work for a "private network install" which is what I need -- using for kubernetes cluster, and want to autoscale control pane without changing etcd). Most of the work was @sagittaros ... but he went on to start putting in other customizations. In fact, I'm probably going to put in some more customizations myself (e.g. override s3 bucket name), but I'd prefer to be closely tracking your repo, as you actually understand the details! :) Thus submitted after merging in the recent changes since his (July) fork.

File: terraform/modules/ignition/ignition.tf (hunk 2 above)
Comment (shaunc) [reply terraform/modules/ignition/ignition.tf]: yes ... seems so -- don't see in release notes but there is discussion in issues.

File: terraform/modules/ignition/ignition.tf (hunk 1 above)
Comment (shaunc) [reply terraform/modules/ignition/ignition.tf]: Oh -- sorry. In fact my "coding style" would have been by default to break both up, as I like to have multiple "strips" of code open in different vertical panes. 
```suggestion
    source = lookup(var.ignition_extra_config, "source", local.blank_ignition_config)
```

**Comment by captn3m0** [comment]
Is the plan to deprecate 0.10 compatibility post this is merged to master? Or will `0.10` branch be maintained with some fixes still?

**Comment by Quentin-M** [comment]
@shaunc @captn3m0 As I am committed to maintaining the core logic of operator (e.g. anything related to the reliability or resiliency of the managed clusters) as I built this to serve the community, a) I am absolutely open to accepting any form of customization / extensions you guys feel necessary, rather than relying on forks that may not get core patches, as well as promoting contributors b) I will be thrilled to make sure we maintain backwards compatibility & backport changes as necessary. I think that maintaining a `master-tf0.10` branch makes sense.

**Comment by shaunc** [comment]
@Quentin-M -- 
1) do you want me to submit a patch for the line break?
2) I also have a patch for moving from terraform-provider-ignition v1.1.0 -> v1.2.0 ... probably best as separate patch (though isn't so extensive: use ".rendered" instead of ".id" in module call)? If so I'll wait to decide if you want to use this as master or as separate branch for the moment.

**Comment by Quentin-M** [comment]
@shaunc Thanks a bunch. Regarding 2/, we have seen some Terraform issues yesterday (that were documented on Red Hat tracker) after our ignition provider in a different project was upgraded - not sure if this is any relevant, or if that is because we were missing some required code changes.

**Comment by Quentin-M** [comment]
@captn3m0 Just pushed `master-tf0.10`

**Pull request merged**